code/modules/mob/living/basic/space_fauna/headslug.dm

```diff
@@ -0,0 +1,97 @@
+/**
+ * ## Headslugs
+ *
+ * Player-controlled slugs that arise from a changeling ability in order to live on in an extremely limited capacity until they can find a suitable corpse to inhabit.
+ */
+/mob/living/basic/headslug
+	name = "headslug"
+	desc = "A small, slug-like creature with a large, gaping maw. It's covered in a thick, slimy mucus."
+	icon_state = "headslug"
+	icon_living = "headslug"
+	icon_dead = "headslug_dead"
+	gender = NEUTER
+	health = 50
+	maxHealth = 50
+	melee_damage_lower = 5
+	melee_damage_upper = 5
+	attack_verb_continuous = "chomps"
+	attack_verb_simple = "chomp"
+	attack_sound = 'sound/weapons/bite.ogg'
+	attack_vis_effect = ATTACK_EFFECT_BITE
+	faction = list(FACTION_CREATURE)
+	obj_damage = 0
+	environment_smash = ENVIRONMENT_SMASH_NONE
+	speak_emote = list("squeaks")
+
+	ai_controller = /datum/ai_controller/basic_controller/headslug
+
+	/// The mind to transfer to our egg when it hatches
+	var/datum/mind/origin
```

Comment:
Why do we have an origin var on headslugs? If it's player controlled then we can just check the mind of the player in the headslug when we need to

Reply:
i think it's a failover? i have no clue why it was here in the first place as i didn't add it, just inherited from the simple-animal framework. might have something to do with lines 78-81 of this file? i'll look into it